### DISCUSSION TITLE
Has --vm-type been removed in 0.5.1?

### DISCUSSION THREAD

**Discussion by anazar2001** [abiosoft/colima#540] [Q&A] [answered]
Can't see this param. Tried to use vmType: vz but it gets automatically set to "" on start. Was this option removed completely? Is it not working at all?

**Comment by rfay** [comment, marked as answer]
It's working fine. But it doesn't exist except with macOS Ventura 13.

Note that you don't change vm types with an existing profile. To experiment, delete your current profile, or create a new one, `colima start vz --vm-type=vz` will start a new `colima-vz` profile with virt type vz.

**Reply by anazar2001** [reply]
Thank you. I am still on Monterey. Does it work with Oracle and RabbitMQ docker containers? Any performance improvements comparing to Monterey/quemu?

**Reply by rfay** [reply]
You'll have to do your own performance comparisons when you can do it. I'm afraid nobody but you has authoritative answers to those questions.

**Reply by anazar2001** [reply]
Mostly worried about Oracle image as colima is the only option I can use at the moment but it starts being unstable after few stop/start cycles.

Usually get errors like below:

deploy-flyway-refdata-1    | #

deploy-flyway-refdata-1    | # A fatal error has been detected by the Java Runtime Environment:

deploy-flyway-refdata-1    | #

deploy-flyway-refdata-1    | #  SIGSEGV (0xb) at pc=0x00007fd067aba848, pid=34, tid=35

deploy-flyway-refdata-1    | #

deploy-flyway-refdata-1    | # JRE version: OpenJDK Runtime Environment AdoptOpenJDK-11.0.11+9 (11.0.11+9) (build 11.0.11+9)

deploy-flyway-refdata-1    | # Java VM: OpenJDK 64-Bit Server VM AdoptOpenJDK-11.0.11+9 (11.0.11+9, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)

deploy-flyway-refdata-1    | # Problematic frame:

deploy-flyway-refdata-1    | # J 31 c1 java.util.HashMap.hash(Ljava/lang/Object;)I java.base@11.0.11 (20 bytes) @ 0x00007fd067aba848 [0x00007fd067aba7c0+0x0000000000000088]

deploy-flyway-refdata-1    | #

deploy-flyway-refdata-1    | # Core dump will be written. Default location: /flyway/core

deploy-flyway-refdata-1    | #

deploy-flyway-refdata-1    | # An error report file with more information is saved as:

deploy-flyway-refdata-1    | # /flyway/hs_err_pid34.log

deploy-flyway-refdata-1    | Compiled method (c1)    3322   31       3       java.util.HashMap::hash (20 bytes)

deploy-flyway-refdata-1    |  total in heap  [0x00007fd067aba610,0x00007fd067abaa48] = 1080

deploy-flyway-refdata-1    |  relocation     [0x00007fd067aba788,0x00007fd067aba7c0] = 56

deploy-flyway-refdata-1    |  main code      [0x00007fd067aba7c0,0x00007fd067aba980] = 448

deploy-flyway-refdata-1    |  stub code      [0x00007fd067aba980,0x00007fd067aba9c0] = 64

deploy-flyway-refdata-1    |  metadata       [0x00007fd067aba9c0,0x00007fd067aba9c8] = 8

deploy-flyway-refdata-1    |  scopes data    [0x00007fd067aba9c8,0x00007fd067aba9e0] = 24

deploy-flyway-refdata-1    |  scopes pcs     [0x00007fd067aba9e0,0x00007fd067abaa30] = 80

deploy-flyway-refdata-1    |  dependencies   [0x00007fd067abaa30,0x00007fd067abaa38] = 8

deploy-flyway-refdata-1    |  nul chk table  [0x00007fd067abaa38,0x00007fd067abaa48] = 16

deploy-flyway-refdata-1    | Could not load hsdis-amd64.so; library not loadable; PrintAssembly is disabled

**Reply by rfay** [reply]
It looks to me like you're probably running an Oracle amd64 image on an arm64 (m1) machine. There are ways that might work, but if you want any kind of stability and predictable behavior, use arm64 images on an arm64 machine.

**Reply by anazar2001** [reply]
I would love to but Oracle is only available for amd64 at the moment. Have you ever seen the error above or should I put it into a separate thread?

**Reply by rfay** [reply]
Every attempt I've ever seen to run oracle mysql on arm64 has ended like what you posted. Have you considered the official https://hub.docker.com/r/mysql/mysql-server/tags (oraclelinux/mysql-8.0 server)?

**Reply by d-led** [reply]
same issue, perhaps: https://github.com/abiosoft/colima/issues/965

**Comment by anazar2001** [comment]
No, I need Oracle as I work for commercial global project which uses Oracle as a main DB. It works if I use x86 architecture Param in Colima but in every 8-10th case I get this error. Stop/start usually fixes the problem. Another problem is speed. It is all very slow. Even though I give Colima 6 cores and 16gig of ram

**Reply by rfay** [reply]
You can try creating a new colima profile with amd64 arch, `arch: amd64` and see if you have better luck. Or you can use an amd64 machine. My own guidance is that you'll not be happy running complex images on a different architecture. Also note that Colima for Ventura has support for Rosetta, which also might get you a little farther down the line, but you'd have to upgrade your machine to use it.

**Reply by anazar2001** [reply]
Unfortunately I can't choose machine or images. Oracle kind of works on M1 Pro with Colima in x86 architecture. Is amd86 any different? What is is the full list of supported architectures in Colima?

**Reply by rfay** [reply]
amd64 == x86
arm64 is mac M1

**Comment by jacksongoode** [comment]
Why is this option restricted to macOS 13, doesn’t 12 also have the virtualization framework?

**Comment by anazar2001** [comment]
Yes, why is it limited to Ventura? In corporate environment it can be difficult to upgrade OS.

**Reply by SuperFola** [reply]
vz support came with MacOS Ventura, thus lima & colima couldn't use it before.
